### PR TITLE
Add copts to framework

### PR DESCRIPTION
**Pull request by jsharpe**
Add `copts` attribute to framework to allow user to add additional compile flags to compile a foreign_cc target with.

### CODE REVIEW

**Review by UebelAndre — CHANGES_REQUESTED** [review]

File: foreign_cc/private/framework.bzl [hunk 1]
```diff
@@ -66,6 +66,11 @@ CC_EXTERNAL_RULE_ATTRIBUTES = {
         mandatory = False,
         default = False,
     ),
+    "copts": attr.string_list(
```
[the remark this answers is not in the document]
Comment (jsharpe) [reply foreign_cc/private/framework.bzl:69]: Yes - its used by the `cc_common` functions and so influences what gets passed as CFLAGS.
I needed to use it in https://github.com/bazelbuild/rules_foreign_cc/pull/588/commits/dd0049a715bc6f4e1278754be8ff0d9f12ac34af# to undefine DEBUG on macOS to successfully get CMake to build.

[the remark this answers is not in the document]
Comment (jsharpe) [reply foreign_cc/private/framework.bzl:69]: I suppose technically we should also add `linkopts` too but I've not yet had a need for that.

[the remark this answers is not in the document]
Comment (UebelAndre) [reply foreign_cc/private/framework.bzl:69]: Might as well add it while we're here. Assuming all that's needed is to plug in the attribute 😄

[the remark this answers is not in the document]
Comment (jsharpe) [reply foreign_cc/private/framework.bzl:69]: `linkopts` was already present. I've added `local_defines` though.

**Comment by jsharpe** [comment]
PTAL @UebelAndre

File: foreign_cc/private/framework.bzl [hunk 2]
```diff
@@ -122,6 +127,11 @@ CC_EXTERNAL_RULE_ATTRIBUTES = {
         mandatory = False,
         default = [],
     ),
+    "local_defines": attr.string_list(
```
Comment (UebelAndre) [inline foreign_cc/private/framework.bzl:130]: Where is `local_defines` used?

**Review by UebelAndre — CHANGES_REQUESTED** [review]
One question.

File: foreign_cc/private/framework.bzl (hunk 2 above)
Comment (jsharpe) [reply foreign_cc/private/framework.bzl:130]: Same as `copts` - its used by the methods in `cc_common` to construct the flags to pass to the `CFLAGS` env variable when setting up the scripts. It allows a user to add a define e.g. `-DNDEBUG` for this target only.

**Review by UebelAndre — APPROVED** [review]

**Comment by UebelAndre** [comment]
This sounds good to me, but if you have a link to the `cc_common` function I think it'd be a good one to have as part of this PR 😄

**Comment by jsharpe** [comment]
This doesn't actually seem to be working now so switching back to draft until I've figured out what's wrong!

**Pull request closed without merging**